waku/v2/protocol/store/client.go

```diff
@@ -0,0 +1,290 @@
+package store
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	libp2pProtocol "github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/libp2p/go-msgio/pbio"
+	"github.com/waku-org/go-waku/logging"
+	"github.com/waku-org/go-waku/waku/v2/peermanager"
+	"github.com/waku-org/go-waku/waku/v2/peerstore"
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+	wpb "github.com/waku-org/go-waku/waku/v2/protocol/pb"
+	"github.com/waku-org/go-waku/waku/v2/protocol/store/pb"
+	"github.com/waku-org/go-waku/waku/v2/timesource"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+)
+
+// StoreQueryID_v300 is the Store protocol v3 identifier
+const StoreQueryID_v300 = libp2pProtocol.ID("/vac/waku/store-query/3.0.0")
+const StoreENRField = uint8(1 << 1)
```

Comment:
It will fail with a protocol not supported error. I think for Status is not a problem since they don't discover store nodes anyway but use instead an specific set, but this might be a problem in nwaku?
cc: @SionoiS